go/mysql/binlog_event_json.go

```diff
@@ -482,10 +482,10 @@ func readOffsetOrSize(data []byte, pos int, large bool) (int, int) {
 }
 
 func readVariableInt(data []byte, pos int) (int, int) {
-	var b byte
+	var b int8
 	var result int
 	for {
-		b = data[pos]
+		b = int8(data[pos])
```

Comment:
Is this fixing a bug? If so, probably merits a unit / regression test.

Reply:
Thank you, eagle-eyed reviewer :)
Yes, it is, and yes, I plan to add one.